R/check.R

```diff
@@ -20,31 +22,32 @@
 #'
 #' try(f(x = 1, y = 2, z = 3))
 #' try(f(x = 1, y = 2, 3, 4, 5))
-check_dots_used <- function(env = parent.frame()) {
-  eval_bare(exit_handler, env)
+check_dots_used <- function(env = parent.frame(), action = abort) {
+  eval_bare(exit_handler(action), env)
   invisible()
 }
 
-check_dots <- function(env = parent.frame()) {
+check_dots <- function(env = parent.frame(), action) {
   if (.Call(ellipsis_dots_used, env)) {
     return(invisible())
   }
 
   proms <- dots(env)
   used <- vapply(proms, promise_forced, logical(1))
 
-  unnused <- names(proms)[!used]
-  stop_dots(
-    message = paste0(length(unnused), " components of `...` were not used."),
-    dot_names = unnused,
-    .subclass = "rlib_error_dots_unnused"
+  unused <- names(proms)[!used]
+  action_dots(
+    action = action,
+    message = paste0(length(unused), " components of `...` were not used."),
+    dot_names = unused,
+    .subclass = "rlib_error_dots_unused",
   )
 }
 
-exit_handler <- bquote(
+exit_handler <- function(action) bquote(
```

Comment:
I don't _think_ making this a functional will change the behavior any, and it seems to work fine in limited testing, but would be useful for someone to verify.

Reply:
`bquote()` is slow and it was on purpose we ran it at build time. It doesn't matter for high level functions but for low level ones it is nice to avoid the overhead. Now that we depend on rlang we could use `expr()`.

Reply:
Also you can use `expr()` directly inside the function, to avoid wrapping into a function.